pyro/ops/arrowhead.py

```diff
@@ -28,12 +28,22 @@ def sqrt(x):
     # NB: the complexity is O(N * head_size^2)
     # ref: https://en.wikipedia.org/wiki/Schur_complement#Background
     Dsqrt = x.bottom_diag.sqrt()
-    B_Dsqrt = B / Dsqrt.unsqueeze(-2)  # shape: head_size x N
-    schur_complement = A - B_Dsqrt.matmul(B_Dsqrt.t())  # complexity: head_size^2 x N
-    # we will decompose schur_complement to U @ U.T (so that the sqrt matrix
-    # is upper triangular) using some `flip` operators:
-    #   flip(cholesky(flip(schur_complement)))
-    top_left = torch.flip(torch.flip(schur_complement, (-2, -1)).cholesky(), (-2, -1))
+
+    for i in range(6):
+        B_Dsqrt = B / Dsqrt.unsqueeze(-2)  # shape: head_size x N
+        schur_complement = A - B_Dsqrt.matmul(B_Dsqrt.t())  # complexity: head_size^2 x N
+        # we will decompose schur_complement to U @ U.T (so that the sqrt matrix
+        # is upper triangular) using some `flip` operators:
+        #   flip(cholesky(flip(schur_complement)))
+        try:
+            top_left = torch.flip(torch.cholesky(torch.flip(schur_complement, (-2, -1))), (-2, -1))
+            break
+        except RuntimeError:
+            B = B * 2 ** (-i - 1)
```

Comment:
Can you confirm that you want super-exponential growth here, not just exponential growth via `B = B / 2`?

Reply:
@martinjankowiak can you confirm this super-exponential backoff is the desired behavior?

Reply:
i can confirm it's what i previously did experiments with

https://github.com/pyro-ppl/pyro/blob/2c709199d5932d48e1d409a9275a4ff67f6e6549/pyro/ops/arrowhead.py#L47

it's probably not the platonic solution but i guess it doesn't hurt... i suspect that in most cases where arrowhead is running into troubles here simplistic regularization will be moderately helpful at best

Reply:
Thanks for reviewing, Fritz! It is not the desired behavior. I will make the change.